### PR TITLE
(Almost) fix MNIST example

### PR DESCRIPTION
**Pull request by fsschneider**
Addresses the bugs described in issue #62.

This should, however, be seen as a hotfix, not as a proper solution.

We currently don't consistently use the `num_X_examples` property for evaluating the model. I would say, ideally, all models should follow the same protocol during `eval_model`, e.g. evaluate on the train set, validation set, and test set.
This would use the `num_X_examples` property, but currently, we are not doing this for most models.

Similarly, we wanted to provide both the `param_shapes` and the `model_params_types` as properties for submitters, but in most workloads, they are not implemented. Ideally, they would be defined in the framework-agnostic base class of each workload.

Lastly, there is the issue that PR #58 introduced an additional `_mask_batch` argument in a couple of places. This is, for example, prominent in the `submission_runner`, which now expects the `data_selection` (a submission function) to also return a `selected_train_mask_batch` variable:

https://github.com/mlcommons/algorithmic-efficiency/blob/c36557c7d8f2411fc316ea33449d0a93640b1252/submission_runner.py#L174-L182

This currently breaks the MNIST example.

We could fix this, by modifying the `data_selection` function of the MNIST submission 
```diff
-return next(input_queue)
+return next(input_queue) + [None]
```
but I am not convinced this is the most elegant solution (a full fix requires a few additional but similar changes).

@znado perhaps you could chip in, but I believe the introduction of the `_mask_batch` arguments currently is also not in line with the API specification we present in the rules: https://github.com/mlcommons/algorithmic-efficiency/blob/main/RULES.md#specification

### CODE REVIEW

**Comment by znado** [comment]
We unfortunately need the mask batch for the GNN workload, so unless we can think of a better way of adding it to the spec I just added it everywhere. For MNIST I implemented the num examples, param types, param shapes, and a nicer solution to the mask batch issue in https://github.com/mlcommons/algorithmic-efficiency/pull/71, so we should go ahead and merge this.

**Comment by fsschneider** [comment]
Closing this as PR #71 provides a nicer solution!
